src/libos/crates/sgx-disk/src/host_disk/sync_io_disk.rs

```diff
@@ -33,23 +34,16 @@ impl SyncIoDisk {
             return Err(errno!(EACCES, "read is not allowed"));
         }
 
+        let fd = self.file.as_raw_fd();
         let (offset, _) = self.get_range_in_bytes(&req)?;
 
-        let mut file = self.file.lock().unwrap();
-        file.seek(SeekFrom::Start(offset as u64))?;
         let read_len = req.access_mut_bufs_with(|bufs| {
-            let mut slices: Vec<IoSliceMut<'_>> = bufs
-                .iter_mut()
-                .map(|buf| IoSliceMut::new(buf.as_slice_mut()))
-                .collect();
-
             // TODO: fix this limitation
             const LINUX_IOVS_MAX: usize = 1024;
-            debug_assert!(slices.len() < LINUX_IOVS_MAX);
+            debug_assert!(bufs.len() < LINUX_IOVS_MAX);
```

Comment:
Put this check in writev and readv. Because that is where buffers are converted to iovecs and thus is where the assertion makes most sense.

Reply:
Done